projects/rero/ng-core/src/lib/menu/menu-widget/menu-widget-prefix-suffix/menu-widget-prefix-suffix.component.html

```diff
@@ -0,0 +1,46 @@
+<!--
+  RERO angular core
+  Copyright (C) 2020 RERO
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU Affero General Public License as published by
+  the Free Software Foundation, version 3 of the License.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<!-- PREFIX -->
+<ng-container *ngIf="item">
+  <ng-container *ngIf="item.getPrefix()">
+    <ng-container *ngIf="item.getPrefix().class">
+      <span [ngClass]="item.getPrefix().class">{{ item.getPrefix().name }}</span>
+    </ng-container>
+    <ng-container *ngIf="!item.getPrefix().class">
+      {{ item.getPrefix().name }}
+    </ng-container>
+  </ng-container>
+
+  <!-- LABEL -->
+  <ng-container *ngIf="item.hasExtra('label-class');">
+    <span [ngClass]="item.getExtra('label-class')">{{ item.getName() }}</span>
+  </ng-container>
+  <ng-container *ngIf="!item.hasExtra('label-class')">
+    {{ item.getName() }}
+  </ng-container>
+
+  <!-- SUFFIX -->
+  <ng-container *ngIf="item.getSuffix()">
+    <ng-container *ngIf="item.getSuffix().class">
+      <span [ngClass]="item.getSuffix().class">{{ item.getSuffix().name }}</span>
+    </ng-container>
+    <ng-container *ngIf="!item.getSuffix().class">
+      {{ item.getSuffix().name }}
+    </ng-container>
+  </ng-container>
```

Comment:
At least use `*ngIf=item.getPrefix() as prefix` to avoid multiple getter calls.

Did you try the simple statement `<span [ngClass]="prefix.class">{{ prefix.name }}</span>` instead of both if ?
It would create a `span` tag without any class if `prefix.class` is null|undefined. In my point of view, it's HTML compilant.

Reply:
done